### DISCUSSION TITLE
Add in firebase multiple Maps

### DISCUSSION THREAD

**Discussion by thallescarvalh0** [SchneiderInfosystems/FB4D#111] [Q&A] [answered]
I have a structure in firebase as follows:

collection/document/collection/document/ collection
Reports  /Sales         /year         /moth       / data

I'm feeding using a database and making a repeating structure:

```
procedure TdmSinc.UpdataGraphics;
var
  Doc: IFirestoreDocument;
begin
  try
    FDQrUserConfig.Close;
    FDQrUserConfig.Open;

    if not AddFirestoreDB(FDQrUserConfigPROJETO_ID.AsString) then
    exit;

   //data 
    FDQrPedFatVendedor.Close;
    FDQrPedFatVendedor.Open;


    FDQrPedFatVendedor.first;
    //Doc := TFirestoreDocument.Create('report');
    while not FDQrPedFatVendedor.Eof do
    begin
       Doc := TFirestoreDocument.Create('report');
       Doc.AddOrUpdateField(TJSONObject.SetMap(FDQrPedFatVendedorSELLER.AsString, [
       TJSONObject.SetDouble('SALES', FDQrPedFatVendedorSALES.AsFloat),
       TJSONObject.SetDouble('REQUESTS', FDQrPedFatVendedorREQUESTS.AsFloat)]));

      fDatabase.InsertOrUpdateDocument(['Reports', 'Sales',
      FDQrPedFatVendedorYEAR.AsString,
      FormatFloat('00', FDQrPedFatVendedorMONTH.AsInteger)], Doc, nil,
      ResponseFirestoreInsertOrUpdate, ErrorFirestore);

      FDQrPedFatVendedor.Next;
    end;
  finally
    FDQrPedFatVendedor.Close;
    FDQrUserConfig.Close;
  end;
end;
```

When querying the "FDQrPedFatVendedor" database, it returns 10 rows and only 1 record in the "map seller" is updated in the firebase base. The map created is represented by the name of the seller and their sales numbers.

**I need to add the 10 rows returned in firebase**

Note: By placing the snippet "Doc := TFirestoreDocument.Create('report');" outside the "while" structure, it can save the 10 records, but if the "MOTH" or "YEAR" are different it does not reset the document "DOC" before saving to firebase. 
In this way, a seller's data is stored in another period

**Comment by SchneiderInfosystems** [comment, marked as answer]
With this structure, you can only store one document per month. Probably, you need more documents, right? 

You could use the following path Year/<Year>/Month/<Month>/Index/<Auto generated ID>.

Please also consult the Firestore DB tutorials to build a suitable DB structure.
